src/Meshes/mesh.ts

```diff
@@ -1302,18 +1302,42 @@ export class Mesh extends AbstractMesh implements IGetSetVerticesData {
      * This method recomputes and sets a new BoundingInfo to the mesh unless it is locked.
      * This means the mesh underlying bounding box and sphere are recomputed.
      * @param applySkeleton defines whether to apply the skeleton before computing the bounding info
+     * @param applyMorph  defines whether to apply the morph target before computing the bounding info
      * @returns the current mesh
      */
-    public refreshBoundingInfo(applySkeleton: boolean = false): Mesh {
+    public refreshBoundingInfo(applySkeleton: boolean = false, applyMorph: boolean = false): Mesh {
         if (this._boundingInfo && this._boundingInfo.isLocked) {
             return this;
         }
 
         const bias = this.geometry ? this.geometry.boundingBias : null;
-        this._refreshBoundingInfo(this._getPositionData(applySkeleton), bias);
+        this._refreshBoundingInfo(this._getPositionData(applySkeleton, applyMorph), bias);
         return this;
     }
 
+    /** @hidden */
+    public _getPositionData(applySkeleton: boolean, applyMorph: boolean): Nullable<FloatArray> {
+        var data = super._getPositionData(applySkeleton, applyMorph);
+
+        if (data && applyMorph && this.morphTargetManager) {

```

Comment:
We still need this when the super function doesn't do it:
```javascript
            data = Tools.Slice(data);
            this._generatePointsArray();
```

Though I'm not sure what is the best way to test this. Maybe recheck `(data && applySkeleton && this.skeleton)`?

Reply:
You're.. I didn't see that ..
What do you think of:
```
        var data = super._getPositionData(applySkeleton, applyMorph);

        if (data) {
            if (!applySkeleton || !this.skeleton) {
                data = Tools.Slice(data);
                this._generatePointsArray();
            }

            if (applyMorph && this.morphTargetManager) {

                // ... 
            }
        }
        return data;
```

Reply:
Yeah, I guess this works. It's a bit dangerous since if someone modifies the logic in the super function, it's not obvious that the child function needs to change, but I don't see a clean way around this. We could maybe pass an out parameter (via an object) to the super function so that we know if the super function has already copied the data, but that's clunky.

Reply:
maybe a  `protected` flag that being set to true every `_generatePointsArray` call and is being reset on each frame? It does feel over-engineered, but I agree with @bghgary that with the suggested implementation we will need to be careful changing the logic in the super function.

Reply:
I don't know if it's relevant, but I am not sure to understand why `skeleton` and `morph target` aren't in the same scope (AbstractMesh). If both were in it, it wouldn't be a problem anymore.

Reply:
@deltakosh?

Reply:
Most a historical reason. We can move both them no problem

Reply:
any update on this?

Reply:
Sorry, not yet (Iam working on BabylonNative for now :p). Next week, I will investigate on the BoundingBox !

Reply:
Hi all,

Here we go, I moved the `morphTargetManager` into `abstractMesh` class and have uniformized the `_getPositionData` function.
What do you think ?

Eric